pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	// Add as per the documentation - https://github.com/go-sql-driver/mysql
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// Provider represents an MySQL DB implementation of the storage.Provider interface
+type Provider struct {
+	dbPath string
+	db     *sql.DB
+	dbs    map[string]*sqlDBStore
+	sync.RWMutex
+}
+
+type sqlDBStore struct {
+	db     *sql.DB
+	dbName string
+}
+
+const (
+	blankDBPathErrMsg         = "DB URL for new mySQL DB provider can't be blank"
+	failToCloseProviderErrMsg = "failed to close provider"
+	//nolint:varcheck deadcode
+	tablePrefix   = "t_"
+	sqlDBNotFound = "no rows"
+)
+
+type result struct {
+	key   string
+	value []byte
+}
+
+// NewProvider instantiates Provider
+func NewProvider(dbPath string) (*Provider, error) {
+	if dbPath == "" {
+		return nil, errors.New(blankDBPathErrMsg)
+	}
+
+	// Example DB Path root:my-secret-pw@tcp(127.0.0.1:3306)/
+	db, err := sql.Open("mysql", dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
+
+	p := &Provider{
+		dbPath: dbPath,
+		db:     db,
+		dbs:    map[string]*sqlDBStore{}}
+
+	return p, nil
+}
+
+// OpenStore opens and returns new db for given name space.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	// creating the database
+	_, err := p.db.Exec("CREATE DATABASE IF NOT EXISTS `name`")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create db: %w", err)
+	}
+
+	// Use is used to select the created database without this DDL operations are not permitted
+	_, err = p.db.Exec("USE `name`")
```

Comment:
same here with the `name` arg

Reply:
actually, I think this is not needed. You should open a new `*sql.DB` and assign it to `sqlDBStore.db` below

Reply:
The new `*sql.DB` should point to the newly created database, of course

Reply:
Some how this doesn't seem to work without the USE Statement. I always get NO Database selected even when i opened the new sql connection with newly created database. Even the official documentation suggest this https://dev.mysql.com/doc/refman/8.0/en/creating-database.html

Reply:
That doc assumes you're still on the same connection. I'm suggesting you open a separate connection for the new database. You'd need to provide a new URL to open that connection.

Reply:
Creating new connection as suggested